### DISCUSSION TITLE
Add ISO 8601 (YYYY-MM-DD etc.) to the locale formatting

### DISCUSSION THREAD

**Discussion by proimage** [craftcms/cms#8971] [Ideas]
### Description
The Formatting Locale in account preferences lists formats by country:
![image](https://user-images.githubusercontent.com/1027523/101788346-a93c2380-3b08-11eb-8df2-db072f1783f5.png)
However, no country has adopted the only format that completely eliminates ambiguity, [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601).

Can we please have a formatting option for ISO 8601?

**Comment by brandonkelly** [comment]
The locale provides a lot of config settings beyond just the date/time formats, including month & weekday names, AM/PM translations, number formatting info, and the localized currency code. So Craft definitely needs to know which locale to pull all that info from.

That said, you could override tho formatter config function from `config/app.php` and swap out the date/time formats with whatever you want:

```php
return [
    'components' => [
        'formatter' => function() {
            $formatter = Craft::$app->getLocale()->getFormatter();
            $formatter->dateTimeFormats = [
                'short' => [
                    'date' => 'y-MM-dd',
                    'time' => 'hh:mm:ss',
                    'datetime' => 'y-MM-ddThh:mm:ss',
                ],
                'medium' => [
                    'date' => 'y-MM-dd',
                    'time' => 'hh:mm:ss',
                    'datetime' => 'y-MM-ddThh:mm:ss',
                ],
                'long' => [
                    'date' => 'y-MM-dd',
                    'time' => 'hh:mm:ss',
                    'datetime' => 'y-MM-ddThh:mm:ss',
                ],
                'full' => [
                    'date' => 'y-MM-dd',
                    'time' => 'hh:mm:ss',
                    'datetime' => 'y-MM-ddThh:mm:ss',
                ],
            ];
            return $formatter;
        },
    ]
];
```

**Comment by proimage** [comment]
Ahh, that's good to know! I wouldn't mind seeing a less general GUI for individual preferences at some point, but for now, a `.php` override should be fine.

Although... how would said override interact with other users' preferences?

**Comment by brandonkelly** [comment]
Hm good question, I guess there’s not really a way to factor that in right now.

**Reply by proimage** [reply]
@brandonkelly Has this been given any more thought?
